udp-native/src/main/java/org/ballerinalang/stdlib/udp/UdpClient.java

```diff
@@ -78,8 +78,9 @@ protected void initChannel(Channel ch) throws Exception {
     // needed for connection oriented client
     private void connect(SocketAddress remoteAddress, SocketAddress localAddress,
                          Future callback) throws  InterruptedException {
-        ChannelFuture channelFuture = clientBootstrap.connect(remoteAddress).sync();
-        channel = clientBootstrap.bind(localAddress).sync().channel();
+        ChannelFuture channelFuture = clientBootstrap.connect(remoteAddress, localAddress).sync();
+//        channel = clientBootstrap.bind(localAddress).sync().channel();
+        channel = channelFuture.sync().channel();
```

Comment:
Instead of invoking sync() here. Can't we do the below ?

```java
        channelFuture.addListener((ChannelFutureListener) future -> {
      
           channel.pipeline().replace(Constants.CONNECTIONLESS_CLIENT_HANDLER,
                Constants.CONNECT_CLIENT_HANDLER, new UdpConnectClientHandler());
           channel.config().setAutoRead(false);

            if (future.isSuccess()) {
                callback.complete(null);
            } else {
                callback.complete(Utils.createSocketError("Can't connect to remote host"));
            }
        });
```

Reply:
+1